src/components/tasks/card/index.tsx

```diff
@@ -442,15 +509,46 @@ const Card: FC<Props> = ({
                                 height={30}
                             />
                         </span>
-                        <span
-                            className={classNames.cardStrongFont}
-                            contentEditable={shouldEdit}
-                            onKeyPress={(e) => handleChange(e, 'assignee')}
-                            role="button"
-                            tabIndex={0}
-                        >
-                            {cardDetails.assignee}
-                        </span>
+                        {shouldEdit ? (
+                            isUserAuthorized && (
+                                <div
+                                    style={{
+                                        position: 'relative',
+                                        display: 'inline-block',
+                                    }}
+                                >
+                                    <input
+                                        ref={inputRef}
+                                        value={assigneeName}
+                                        className={classNames.cardStrongFont}
+                                        onKeyDown={(e) => {
+                                            handleChange(e, 'assignee');
+                                        }}
+                                        onChange={(e) => {
+                                            handleAssignment(e);
+                                            debounce(
+                                                fetchUsers,
+                                                400
+                                            )(e.target.value);
+                                        }}
+                                        role="button"
+                                        tabIndex={0}
+                                    />
+
+                                    {showSuggestion && (
+                                        <SuggestionBox
+                                            suggestions={suggestions}
+                                            onClickName={handleClick}
+                                            loading={isLoadingSuggestions}
+                                        />
+                                    )}
+                                </div>
```

Comment:
Suggestion: we can take this out into a separate component it will improve the code readability